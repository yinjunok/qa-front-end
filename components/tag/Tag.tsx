import * as React from 'react';
import * as css from './styles.less';

type TagProps = React.FunctionComponent<React.HTMLAttributes<HTMLSpanElement>>;

const Tag: TagProps = ({ children, className, ...props }) => {
  return (
    <span className={`${css.tag} ${className}`} {...props}>{children}</span>
  );
}

Tag.defaultProps = {
  className: '',
}

export default Tag;