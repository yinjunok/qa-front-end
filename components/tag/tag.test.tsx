import renderer from 'react-test-renderer';
import Tag from './Tag';

describe('test ui', () => {
  it('snapshoot', () => {
    const tag1 = renderer.create(
      <Tag>标签</Tag>
    );

    const tag2 = renderer.create(
      <Tag className='tag'>标签</Tag>
    );

    expect(tag1.toJSON()).toMatchSnapshot();
    expect(tag2.toJSON()).toMatchSnapshot();
  });
});
