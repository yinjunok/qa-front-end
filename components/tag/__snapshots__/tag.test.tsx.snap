// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ui snapshoot 1`] = `
<span
  className="tag "
>
  标签
</span>
`;

exports[`test ui snapshoot 2`] = `
<span
  className="tag tag"
>
  标签
</span>
`;
